src/adapter/src/rbac.rs

```diff
@@ -167,7 +176,7 @@ pub fn check_plan(
     }
 
     // Obtain all roles that the current session is a member of.
-    let role_membership = catalog.collect_role_membership(role_id);
+    let role_membership = catalog.collect_role_membership(session_role_id);
```

Comment:
Clarifying question for my own sake, why do we use the `session_role_id` here and not the `current_role_id`?